### DISCUSSION TITLE
Incremental Static Regeneration aggressively  regenerating pages on Link tag?

### DISCUSSION THREAD

**Discussion by pkellner** [vercel/next.js#20780] [Help]
I seem to have noticed that if I have a bunch of links on a page, that as soon as the browser visits that page, all those pages are regenerated.  I don't see an option to disable that (or any discussion of that happening).  I noticed it because my server got really busy doing regenerations when only one page had been requested.

Please clarify behavior on links and ISR.

This is the code that caused the regens I believe.

```
<div key={speakerRec.id}>
              <Link
                href={{
                  pathname: '/presenter/[year]/[nameSlug]',
                  query: { nameSlug: speakerRec.nameSlug, year: year },
                }}
              >
                <a>{speakerRec.nameSlug}</a>
              </Link>
            </div>
```

**Comment by danielr18** [comment]
@pkellner The prefetch requests regenerate the pages, there was no way to prevent this last time I went through the source code.

I created a Link wrapper to add `prefetch={false}` to all links.

```
const Link: React.FC<LinkProps> = ({ href, children, ...props }) => {
  return (
    <NextLink href={href} prefetch={false} {...props}>
      {children}
    </NextLink>
  );
};
```
